src/Bridge/Laravel/Console/Commands/SearchIndexCleanCommand.php

```diff
@@ -0,0 +1,55 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\Search\Bridge\Laravel\Console\Commands;
+
+use Illuminate\Contracts\Container\Container as ContainerInterface;
+use LoyaltyCorp\Search\Interfaces\HandlerInterface;
+use LoyaltyCorp\Search\Interfaces\IndexerInterface;
+
+class SearchIndexCleanCommand extends SearchIndexCommand
+{
+    /**
+     * @var \LoyaltyCorp\Search\Interfaces\IndexerInterface
+     */
+    private $indexer;
+
+    /**
+     * @var \LoyaltyCorp\Search\Interfaces\HandlerInterface[]
+     */
+    private $searchHandlers = [];
+
+    /**
+     * SearchIndexClean constructor.
+     *
+     * @param \Illuminate\Contracts\Container\Container $container
+     * @param \LoyaltyCorp\Search\Interfaces\IndexerInterface $indexer
+     */
+    public function __construct(ContainerInterface $container, IndexerInterface $indexer)
+    {
+        $this->description = 'Remove any indices deriving from search handlers that are unused';
+        $this->signature = 'search:index:clean';
+
+        $this->indexer = $indexer;
+
+        parent::__construct($container);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(): void
+    {
+        parent::handle();
+
+        $this->indexer->clean($this->searchHandlers);
```

Comment:
I'd like to see this command output a list of things that actually happened (for this one, it'd be nice if there was a "Hey i'm about to do this stuff, do you confirm?")